### PR TITLE
Adding me.sergiotarxz.JapaChar

### PR DESCRIPTION
**Pull request by sergiotarxz**
### Please confirm your submission meets all the criteria

This is an app to memorize characters japanese romanji representation. 

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: me.sergiotarxz.JapaChar.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+runtime-version: "46"
+sdk: "org.gnome.Sdk"
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=xdg-config/.japachar
+command: '/app/JapaChar/run.sh'
+modules: 
+  - name: "perle"
```
Comment (hfiguiere) [inline me.sergiotarxz.JapaChar.yml:13]:
```suggestion
  - name: "perl"
```

File: me.sergiotarxz.JapaChar.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          - 'set -e'
+          - 'cp -vr JapaCharSource ${FLATPAK_DEST}/JapaChar'
+          - 'mkdir -pv /app/share/applications'
+          - 'cp -v JapaCharSource/me.sergiotarxz.JapaChar.desktop /app/share/applications/'
+          - 'cp -v JapaCharSource/run.sh /app/bin/japachar'
+          - 'mkdir -pv /app/share/icons/hicolor/scalable/apps/'
+          - 'cp -pv JapaCharSource/me.sergiotarxz.JapaChar.svg /app/share/icons/hicolor/scalable/apps/'
+        dest: "japachar"
+        dest-filename: "install.sh"
+        type: "script"
```
Comment (hfiguiere) [inline me.sergiotarxz.JapaChar.yml:99]:
```suggestion
    build-commands:
          - 'cp -vr JapaCharSource ${FLATPAK_DEST}/JapaChar'
          - 'install -Dm644 JapaCharSource/me.sergiotarxz.JapaChar.desktop -t /app/share/applications/'
          - 'install -Dm755 JapaCharSource/run.sh /app/bin/japachar'
          - 'install -Dm644 JapaCharSource/me.sergiotarxz.JapaChar.svg -t /app/share/icons/hicolor/scalable/apps/'
    sources: 
      - type: "git"
        url: 'https://git.owlcode.tech/sergiotarxz/JapaChar'
        tag: "v0.01"
        dest: 'JapaCharSource'
```

no need to make things complicated.

File: me.sergiotarxz.JapaChar.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+branch: "main"
+runtime: "org.gnome.Platform"
+runtime-version: "46"
+sdk: "org.gnome.Sdk"
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=xdg-config/.japachar
+command: '/app/JapaChar/run.sh'
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply me.sergiotarxz.JapaChar.yml:11]:
```suggestion
command: japachar
```

File: me.sergiotarxz.JapaChar.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
+id: "me.sergiotarxz.JapaChar"
+branch: "main"
+runtime: "org.gnome.Platform"
+runtime-version: "46"
+sdk: "org.gnome.Sdk"
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=xdg-config/.japachar
```
Comment (hfiguiere) [inline me.sergiotarxz.JapaChar.yml:10]: why do you need this? In general accessing `xdg-config` in incorrect and not allowed.

File: me.sergiotarxz.JapaChar.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
+id: "me.sergiotarxz.JapaChar"
+branch: "main"
```
Comment (hfiguiere) [inline me.sergiotarxz.JapaChar.yml:2]:
```suggestion
```

never a flatpak branch

also invalid value.

File: me.sergiotarxz.JapaChar.yml [hunk 6]
```diff
@@ -0,0 +1,99 @@
+id: "me.sergiotarxz.JapaChar"
```
Comment (hfiguiere) [inline me.sergiotarxz.JapaChar.yml:1]: `sergiotarxz.me` is 404 not found.

Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml:1]: It is no longer 404. I thought only by having the domain was enough.

File: me.sergiotarxz.JapaChar.yml (hunk 5 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml:2]: Can I remove this option? I cannot find documentation for that.

File: me.sergiotarxz.JapaChar.yml (hunk 6 above)
Comment (hfiguiere) [reply me.sergiotarxz.JapaChar.yml:1]: and how would we verify otherwise?

File: me.sergiotarxz.JapaChar.yml (hunk 4 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml:10]: To preserve progress of users between reinstallations.

Comment (hfiguiere) [reply me.sergiotarxz.JapaChar.yml:10]: Are you sure that's the right option.

you probably want `--persist=.japachar` instead.

Ideally it should follow xdg standard and use `$XDG_CONFIG_HOME` to locate the directory.

File: me.sergiotarxz.JapaChar.yml (hunk 5 above)
Comment (hfiguiere) [reply me.sergiotarxz.JapaChar.yml:2]: what's unclear?

File: me.sergiotarxz.JapaChar.yml (hunk 4 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml:10]: I will try that, thank you, it is my first time doing this.

File: me.sergiotarxz.JapaChar.yml (hunk 5 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml:2]: I do not understand your message, I do not know what a "flatpak branch" is.

I understand every other message you sent, but I am having trouble understanding this.

File: me.sergiotarxz.JapaChar.yml (hunk 4 above)
Comment (hfiguiere) [reply me.sergiotarxz.JapaChar.yml:10]: so you didn't test it.....

File: me.sergiotarxz.JapaChar.yml (hunk 1 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml]: Fixed

File: me.sergiotarxz.JapaChar.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml]: Fixed

File: me.sergiotarxz.JapaChar.yml (hunk 2 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml]: Fixed

File: me.sergiotarxz.JapaChar.yml (hunk 5 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml]: I have deleted the option entirely, I hope that's enough, reopen if it isn't please.

File: me.sergiotarxz.JapaChar.yml (hunk 4 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml]: Done.

Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml]: @hfiguiere Yes, I tested with flatpak-builder and running the app.

File: me.sergiotarxz.JapaChar.yml (hunk 5 above)
Comment (hfiguiere) [reply me.sergiotarxz.JapaChar.yml]: that was the suggestion. still don't get what was unclear.

File: me.sergiotarxz.JapaChar.yml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+runtime: "org.gnome.Platform"
+runtime-version: "46"
+sdk: "org.gnome.Sdk"
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --persist=.japachar
```
Comment (hfiguiere) [inline me.sergiotarxz.JapaChar.yml:9]: Reading this https://git.owlcode.tech/sergiotarxz/JapaChar/src/branch/main/lib/JapaChar/DB.pm#L41

it's likely to be unnecessary. But again you should know better that I do.

File: me.sergiotarxz.JapaChar.yml (hunk 5 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml]: Ah, ok, sorry, I did not got that was a suggestion.

File: me.sergiotarxz.JapaChar.yml (hunk 7 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml:9]: What you say is true, I just tested, removing it..

**Comment by sergiotarxz** [comment]
I have rebased.

**Comment by hfiguiere** [comment]
bot, build me.sergiotarxz.JapaChar

**Comment by hfiguiere** [comment]
the appstream file is missing.

**Comment by hfiguiere** [comment]
sorry wrong button

**Comment by sergiotarxz** [comment]
I have fixed every linter error, I think should be ok now.

**Comment by sergiotarxz** [comment]
bot, build me.sergiotarxz.JapaChar

File: me.sergiotarxz.JapaChar.yml [hunk 8]
```diff
@@ -0,0 +1,90 @@
+id: "me.sergiotarxz.JapaChar"
+runtime: "org.gnome.Platform"
+runtime-version: "46"
+sdk: "org.gnome.Sdk"
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
```
Comment (bbhtt) [inline me.sergiotarxz.JapaChar.yml:8]:
```suggestion
  - --share=ipc
  - --device=dri
  - --socket=fallback-x11
  - --socket=wayland
```

**Comment by bbhtt** [comment]
The png produced from you svg is blank. Can you please fix the icon? You can check the appstream produced icons by installing the build and going to `/home/test/.local/share/flatpak/app/me.sergiotarxz.JapaChar/current/active/files/share/app-info/icons`

File: me.sergiotarxz.JapaChar.yml (hunk 8 above)
Comment (sergiotarxz) [reply me.sergiotarxz.JapaChar.yml]: Ready

**Comment by bbhtt** [comment]
As a result of this GNOME software shows up with effectively a blank icon

image

**Comment by sergiotarxz** [comment]
Hi, @bbhtt I tried setting the icons to their final png to avoid this error but when I do:
```
      - 'install -Dm644 JapaCharSource/me.sergiotarxz.JapaChar-64.png /app/share/icons/hicolor/64x64/apps/me.sergiotarxz.JapaChar.png'
      - 'install -Dm644 JapaCharSource/me.sergiotarxz.JapaChar-128.png /app/share/icons/hicolor/128x128/apps/me.sergiotarxz.JapaChar.png'   
```
I get this cryptic error:

```
Commit: 73f89e8b0e39051ae9422f5dc105a5178762478c760b806313df48add89e9b29
Metadata Total: 675
Metadata Written: 1
Content Total: 693
Content Written: 0
Content Bytes Written: 0 (0 bytes)
Installing app/me.sergiotarxz.JapaChar/x86_64/master
Error: Failed to install me.sergiotarxz.JapaChar: Is a directory
Install failed: Child process exited with code 1
```

The one which fails is the 128x128 one, the 64x64 one works fine.

Do you think it could be enough only with a 64x64 icon? Or what I could do to prevent this error?

**Comment by sergiotarxz** [comment]
bot, build me.sergiotarxz.JapaChar

**Comment by sergiotarxz** [comment]
I have tried the recommended file size of flathub 256x256 and it installed ok so I won't worry anymore.

**Comment by bbhtt** [comment]
Icon properly shows up now but the all black icon does not look very good. Anyway it's not a blocker but maybe you fix it later.

**Comment by sergiotarxz** [comment]
I would say to  leave icon rework for later, mostly because I am not an icon designer and this is probably the best I can currently do, I am hoping that someone which contributes to the app reworks it.

**Comment by sergiotarxz** [comment]
Hi, now that the pull request is marked as ready...
What should I do?
Can I simply type /merge or something else must be done on your end?

**Comment by bbhtt** [comment]
We will merge it, in time. Merges are batched.

**Comment by sergiotarxz** [comment]
@bbhtt If it is like this then I will fix the icon, I made a rework, but did not want to stop a merge near to happen and just hoped to fix it later.

**Comment by sergiotarxz** [comment]
bot, build me.sergiotarxz.JapaChar

**Comment by sergiotarxz** [comment]
Improved icon with success.

**Comment by sergiotarxz** [comment]
bot, build me.sergiotarxz.JapaChar

File: me.sergiotarxz.JapaChar.yml [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    build-commands:
+      - 'cp -vr JapaCharSource ${FLATPAK_DEST}/JapaChar'
+      - 'install -Dm644 JapaCharSource/me.sergiotarxz.JapaChar.desktop -t /app/share/applications/'
+      - 'install -Dm755 JapaCharSource/run.sh /app/bin/japachar'
+      - 'install -Dm644 JapaCharSource/me.sergiotarxz.JapaChar-256.png /app/share/icons/hicolor/256x256/apps/me.sergiotarxz.JapaChar.png'
+      - 'install -Dm644 JapaCharSource/me.sergiotarxz.JapaChar.metainfo.xml -t /app/share/metainfo/' 
+    sources: 
+      - type: "git"
+        url: 'https://git.owlcode.tech/sergiotarxz/JapaChar'
+        tag: "v0.09"
```
Comment (hfiguiere) [inline me.sergiotarxz.JapaChar.yml:90]: there should be a commit here as well.

**Comment by hfiguiere** [comment]
> I triggered a new build because I found a bug upstream that didn't allow to continue studying once you had more than 100 score on every character (No new character to learn) and fixed it.

The problem here is that if the new build fix the bug that mean you re-tagged. This is not good. NEVER re-tag. Create a new version. And make sure there is a commit along with a tag in the manifest.

Retagging is bad and is not specific to flatpak.

**Comment by sergiotarxz** [comment]
> > I triggered a new build because I found a bug upstream that didn't allow to continue studying once you had more than 100 score on every character (No new character to learn) and fixed it.
> 
> The problem here is that if the new build fix the bug that mean you re-tagged. This is not good. NEVER re-tag. Create a new version. And make sure there is a commit along with a tag in the manifest.
> 
> Retagging is bad and is not specific to flatpak.

I was thinking it was better than releasing a new version for every bug when it is not even in flathub yet, I will keep in mind, I am adding a commit to the `.yml` to avoid using further that practice.

**Comment by hfiguiere** [comment]
> I was thinking it was better than releasing a new version for every bug 

No it's not.

**Comment by sergiotarxz** [comment]
bot, build me.sergiotarxz.JapaChar

**Comment by sergiotarxz** [comment]
bot, build me.sergiotarxz.me because I polished the first use performance.

**Comment by sergiotarxz** [comment]
bot, build me.sergiotarxz.JapaChar because I polished the first use performance.

**Comment by bbhtt** [comment]
Let me know if you are done fixing, I'll mark it as ready once it is final from your side.

**Comment by sergiotarxz** [comment]
@bbhtt I am not going to fix anything else.

**Comment by bbhtt** [comment]
bot, build me.sergiotarxz.JapaChar

**Comment by sergiotarxz** [comment]
Everything got /merge d but my project did something happen?

**Comment by sergiotarxz** [comment]
Do I have to wait to the next batch?

**Pull request closed without merging**